### PR TITLE
python38Packages.jsonschema: depend on pkgutil-resolve-name

### PR DESCRIPTION
**Pull request by yajo**
###### Description of changes

python38Packages.jsonschema: depend on pkgutil-resolve-name

python38Packages.jsonschema: depend on pkgutil-resolve-name

The missing dependency was making the build fail.

See https://github.com/python-jsonschema/jsonschema/blob/d35f2c258a8a4fed7533d73e9fe3f3ec36c579e8/pyproject.toml#L42

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

@moduon MT-1075

### CODE REVIEW

File: pkgs/development/python-modules/pkgutil-resolve-name/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{ buildPythonPackage
+, fetchPypi
+, lib
+, nix-update-script
+, pythonOlder
+}:
+buildPythonPackage rec {
+  pname = "pkgutil_resolve_name";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pkgutil-resolve-name/default.nix:8]:
```suggestion
  pname = "pkgutil-resolve-name";
```

File: pkgs/development/python-modules/pkgutil-resolve-name/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  version = "1.3.10";
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-NX1snmp1VlPP14iTgXwIU682XdUeyX89NYqBk3O70XQ=";
+  };
+  format = "flit";
+  disabled = pythonOlder "3.7";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pkgutil-resolve-name/default.nix:15]:
```suggestion
  version = "1.3.10";
  format = "flit";

  disabled = pythonOlder "3.7";

  src = fetchPypi {
    pname = "pkgutil_resolve_name";
    inherit version;
    hash = "sha256-NX1snmp1VlPP14iTgXwIU682XdUeyX89NYqBk3O70XQ=";
  };
```

**Review by SuperSandro2000 — COMMENTED** [review]
This should target master.

Also why do we need this? 3.8 is EOL.

**Comment by yajo** [comment]
All comments attended.

> Also why do we need this? 3.8 is EOL.

AFAICS in https://endoflife.date/python, 3.8 will still be supported for more than 1 year.

**Comment by SuperSandro2000** [comment]
We are not following the upstream compiler end of life cycle but our own to reduce the major maintenance overhead of maintaining multiple versions of packages to support the entire range of python versions. We stopped maintaing python38Packages in #152307 1.5 years ago.

**Comment by yajo** [comment]
Ah I see... Well, I need this patch and I'm gonna use it. I can maintain it downstream if you prefer, but I thought that it'd be useful here just in case somebody else needs it. It's up to you! 😊

File: pkgs/development/python-modules/pkgutil-resolve-name/default.nix [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  version = "1.3.10";
+  format = "flit";
+
+  disabled = pythonOlder "3.7";
+
+  src = fetchPypi {
+    pname = "pkgutil_resolve_name";
+    inherit version;
+    hash = "sha256-NX1snmp1VlPP14iTgXwIU682XdUeyX89NYqBk3O70XQ=";
+  };
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pkgutil-resolve-name/default.nix:18]:
```suggestion
  };

  # has no tests
  doCheck = false;
```

**Review by SuperSandro2000 — APPROVED** [review]
we talked now so much about this we might as well include it

**Comment by yajo** [comment]
So, any chances to get a merge?

**Comment by SuperSandro2000** [comment]
@ofborg build python310Packages.pkgutil-resolve-name

**Pull request merged**